seer_pas_sdk/core/unsupported.py

```diff
@@ -1448,3 +1448,348 @@ def _get_msdataindex_path(self, display_path: list):
             )
 
         return result
+
+    def get_search_data(
+        self, analysis_id: str, analyte_type: str, rollup: str, engine: str
+    ):
+        """
+        Get search data for a given analysis ID.
+        Args:
+            analysis_id (str): ID of the analysis.
+            analyte_type (str): Type of analyte. Must be either 'protein', 'peptide', precursor.
+            rollup (str): Rollup type. Must be either 'np' or 'panel'.
+            engine (str): Search engine. Supported engines are: raw, diann, median, median80, pepcal.
+
+        Returns:
+            pd.DataFrame: A dataframe with each row containing the following columns:
+                            'msrun_id', 'sample_id', 'nanoparticle' (if rollup is 'np'), 'protein_group', 'peptide' (for peptide),
+                            'intensity_log10', 'q_value' (for protein), 'rt' and 'irt' (for peptide)
+        """
+        # 1. Get msrun data for analysis
+        samples = self.find_samples(analysis_id=analysis_id)
+        sample_name_to_id = {s["sample_name"]: s["id"] for s in samples}
+        # for np rollup, a row represents an msrun
+        msruns = self.find_msruns(sample_ids=sample_name_to_id.values())
+        file_to_msrun = {
+            os.path.basename(msrun["raw_file_path"]).split(".")[0]: msrun
+            for msrun in msruns
+        }
+        sample_to_msrun = {msrun["sample_id"]: msrun for msrun in msruns}
+
+        # for panel rollup, a row represents a sample
+
+        # 2. Get search results
+        # pull the np/panel file, or report.tsv for precursor mode
+        search_results = self.get_search_result(
+            analysis_id=analysis_id,
+            analyte_type=analyte_type,
+            rollup=rollup,
+        )
+        if analyte_type in ["protein", "peptide"]:
+            intensity_column = None
+            if engine.casefold() == "raw":
+                intensity_column = (
+                    "Intensities Log10"
+                    if "Intensities Log10" in search_results.columns
+                    else "Intensity (Log10)"
+                )
+            elif engine.casefold() == "diann":
```

Comment:
Is there a specific reason for a `casefold()`? It's not done for the other params.

```suggestion
            elif norm_method == "engine":
```